CHANGES.md

```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
++ macros: Improve error reporting on invalid trait implementations
```

Comment:
Can you add a UI test that demonstrates the improvement?

Reply:
Old error message: 13fdd5fd2b8587e8febdca60e9ec7a5ab0a60820
New error message: 9b5f168508c58c6f4919544da55133a2f0d9375c

The new error reporting encourages rustc to further analyze the code. It is not better in every case, but I hope that it will help rust-analyzer to offer more completions in the future due to less code changed by the macro.